cmd/podman/main.go

```diff
@@ -29,13 +30,28 @@ import (
 	"github.com/spf13/cobra"
 )
 
+const PodmanSh = "podmansh"
+
 func main() {
 	if reexec.Init() {
 		// We were invoked with a different argv[0] indicating that we
 		// had a specific job to do as a subprocess, and it's done.
 		return
 	}
 
+	if filepath.Base(os.Args[0]) == PodmanSh ||
+		filepath.Base(os.Args[0][1:]) == PodmanSh {
+		shell := os.Args[0]
+		if string(shell[0]) == "-" {
```

Comment:
This if statement should not be needed, TrimPrefix() will only trim when necessary otherwise return the string as it.

Reply:
updated. PTAL